tests/unit/checkmate/checker/url/url_haus_test.py

```diff
@@ -0,0 +1,89 @@
+from unittest.mock import sentinel
+
+import pytest
+from h_matchers import Any
+from httpretty import httprettified, httpretty
+from pkg_resources import resource_string
+
+from checkmate.checker.url import URLHaus
+from checkmate.checker.url.reason import Reason
+
+
+class TestURLHaus:
+    def test_check_url_with_hits(self, URLHausRule):
+        URLHausRule.find_matches.return_value.count.return_value = 2
+
+        response = URLHaus(sentinel.db_session).check_url(["hex_hash"])
+
+        assert response == Any.generator.containing([Reason.MALICIOUS]).only()
+        # The above will exchaust the generator and allow us to make assertions
+        # about the call
+        URLHausRule.find_matches.assert_called_once_with(
+            sentinel.db_session, ["hex_hash"], limit=1
+        )
+
+    def test_check_url_with_no_hits(self, URLHausRule):
+        URLHausRule.find_matches.return_value.count.return_value = 0
+
+        response = URLHaus(sentinel.db_session).check_url(["hex_hash"])
+
+        assert response == Any.generator.containing([]).only()
+
+    @httprettified
+    def test_reinitialise_db(self, URLHausRule):
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://urlhaus.abuse.ch/downloads/csv/",
+            body=self.read_fixture("csv.txt.zip"),
+        )
+
+        URLHaus(sentinel.db_session).reinitialise_db()
+
+        URLHausRule.truncate.assert_called_once_with(sentinel.db_session)
+        self.assert_expected_sync(URLHausRule)
+
+    def test_partial_update(self, URLHausRule):
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://urlhaus.abuse.ch/downloads/csv_recent/",
+            body=self.read_fixture("csv.txt"),
+        )
+
+        URLHaus(sentinel.db_session).update_db()
+
+        URLHausRule.truncate.assert_not_called()
+        self.assert_expected_sync(URLHausRule)
+
+    def assert_expected_sync(self, URLHausRule):
+        URLHausRule.bulk_upsert.assert_called_once_with(
+            session=sentinel.db_session, values=Any.generator()
+        )
+        assert URLHausRule.updated_values == [
+            {
+                "hash": "7d93a7a785da3bb7fc67b08cda3368745eb7cf6155e4d8b26415680e69a3f5c6",
+                "id": "904859",
+                "rule": "180.112.189.18/bin.sh",
+            },
+            {
+                "hash": "f1991c232fda31acdfb50bf118458ddfd31140649218f4774f9c98e50317a59c",
+                "id": "904858",
+                "rule": "117.194.161.173/bin.sh",
+            },
```

Comment:
I was thinking about using the hash function here instead of hardcoded hashes, but... :shrug: good enough?